source/MaterialXGenOsl/OslSyntax.cpp

```diff
@@ -217,12 +217,12 @@ class OSLMatrix3TypeSyntax : public AggregateTypeSyntax
         for (size_t i = 0; i<values.size(); i++)
         {
             ss << values[i] << ", ";
-            if ((i != 0) && (i % 3 == 0))
+            if ((i+1) % 3 == 0)
```

Comment:
Is this change equivalent to what was there before?
i+1 % 3 != i % 3.

Reply:
No, they aren't the same, the previous logic wasn't quite right.